File: src/main/java/com/dashboard/interfaces/serviceInterfaces/UserServiceInterface.java
package com.dashboard.interfaces.serviceInterfaces;

import com.dashboard.models.JPA.User;

import java.util.List;

public interface UserServiceInterface {
    void save(User user);
    void delete(Integer userId);
    void update(Integer userId, User user);
    boolean exist(Integer userId);
    List<User> getAllUsers();
    User getUserById(Integer userId);
}

File: src/main/kotlin/dashboard/interfaces/serviceInterfaces/UserServiceInterface.kt
package dashboard.interfaces.serviceInterfaces

import dashboard.models.jpa.User

interface UserServiceInterface {
    fun save(user: User)
    fun delete(userId: Int)
    fun update(userId: Int, user: User)
    fun exist(userId: Int): Boolean
    val allUsers: List<User>
    fun getUserById(userId: Int): User
}